### PR TITLE
feat: publish the name of new node to cluster when node start

### PR DESCRIPTION
**Pull request by themanforfree**
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)
 publish the name of new node to cluster when node start
* what changes does this pull request make?
When a node starts, it publishes its name to the cluster by publish RPC, 
* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)
no

### CODE REVIEW

File: xline/src/server/xline_server.rs [hunk 1]
```diff
@@ -305,6 +314,7 @@ impl XlineServer {
         MaintenanceServer<S>,
         ClusterServer,
         Arc<CurpServer<S>>,
+        Arc<CurpClient>,
```
Comment (iGxnon) [inline xline/src/server/xline_server.rs:317]: Could we call `publish` in `init_servers`?

File: curp/src/server/raw_curp/mod.rs [hunk 2]
```diff
@@ -338,6 +338,20 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
...
+    pub(super) fn handle_publish(&self, req: PublishRequest) -> Result<(), CurpError> {
+        debug!("{} gets publish with propose id {}", self.id(), req.id());
+        let st_r = self.st.read();
+        if st_r.role != Role::Leader {
+            return Err(CurpError::Redirect(st_r.leader_id, st_r.term));
+        }
+        let mut log_w = self.log.write();
+        let entry = log_w.push(st_r.term, req)?;
+        debug!("{} gets new log[{}]", self.id(), entry.index);
+        self.entry_process(&mut log_w, entry, false);
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply curp/src/server/raw_curp/mod.rs:351]: Yes, this request only affects the server name that the user sees, all other requests are not related to it,so it does not need to conflict with other requests

File: curp/src/server/raw_curp/mod.rs [hunk 3]
```diff
@@ -563,7 +577,10 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
                         matches!(c.change_type(), ConfChangeType::Remove)
                             && c.node_id == candidate_id
                     }),
-                    EntryData::Empty(_) | EntryData::Command(_) | EntryData::Shutdown(_) => false,
+                    EntryData::Empty(_)
+                    | EntryData::Command(_)
+                    | EntryData::Shutdown(_)
+                    | EntryData::SetName(_, _, _) => false,
                 });
```
[the remark this answers is not in the document]
Comment (themanforfree) [reply curp/src/server/raw_curp/mod.rs:584]: It's not very complicated here. There is still a possibility of adding new EntryData in the future, and the processing here may be different, so I think it's better to match all members explicitly

File: xline/src/server/xline_server.rs (hunk 1 above)
Comment (themanforfree) [reply xline/src/server/xline_server.rs:317]: No, the RPC server has not been started at this stage

**Review by Phoenix500526 — APPROVED** [review]
LGTM

**Review by iGxnon — APPROVED** [review]
LGTM

**Pull request merged**